### PR TITLE
[WIP] Update UG 

### PR DESCRIPTION
**Pull request by simonteozw**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -1,225 +1,352 @@
...
+Format: `T add n/NAME d/DATE a/LOCATION w/WEIGHTAGE r/REMARK`
 
-Examples: `T add n/Assignment 1 due d/20/11/2020 a/NUS w/20 r/Focus on Chapters 1-3`
+Examples: `T add n/Assignment 1 d/20/11/2020 a/NUS w/20 r/Focus on Chapters 1-3`
+
+Remark: The `DATE` provided must be in the form `dd/MM/yyyy`. The `WEIGHTAGE` provided must be a number.
 
 ### Edit
-Description: Edits the detail of a task.
+Description: Edit the specified task.
```
Comment (justweihong) [inline docs/UserGuide.md]: Maybe can mention that the task is specified by index?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -1,225 +1,352 @@
...
+TrackIt@NUS data is saved in the hard disk automatically after any command that changes the data. There is no need to
+ save manually.
 
 --------------------------------------------------------------------------------------------------------------------
 
-## FAQ
+## FAQ <a name="faq"></a>
 
 **Q**: How do I transfer my data to another Computer?<br>
 **A**: Install the app in the other computer and overwrite the empty data file it creates with the file that contains the data of your previous AddressBook home folder.
```
Comment (justweihong) [inline docs/UserGuide.md]: Replace AddressBook with TrackIT@NUS?

File: docs/UserGuide.md [hunk 3]
```diff
@@ -1,225 +1,352 @@
...
-Examples: `M add m/CS1231S n/Discrete Structures d/Introductory mathematical tools required for computer science`
+Format: `M add m/MODULE_CODE n/NAME d/DESCRIPTION`
+
+Examples: `M add m/CS1231S n/Discrete Structures d/Introductory mathematical tools required for Computer Science`
 
 ### Edit
-For: Edit a module.
+For: Edit the specified module.
+
+Format: `M edit INDEX [m/MODULE_CODE] [n/NAME] [d/DESCRIPTION]`
```
Comment (UncleGrandpa925) [inline docs/UserGuide.md]: Hmm should we edit the module by index? or should it be index?

File: docs/UserGuide.md [hunk 4]
```diff
@@ -1,225 +1,352 @@
...
 
-Examples: `T info n/CS2100 midterm`
+Remark: The `INDEX` provided must be positive and cannot be larger than the length of the Module list.
 
 ### Add
-Description: Adds a task to TrackIt@NUS
+Description: Add a new task.
 
-Format: `T add n/{task name} d/{task date} a/{location} w/{weightage} r/{remark}`
+Format: `T add n/NAME d/DATE a/LOCATION w/WEIGHTAGE r/REMARK`
```
Comment (UncleGrandpa925) [inline docs/UserGuide.md]: We haven't agreed on whether a task must be associated with a module or not right?

File: docs/UserGuide.md [hunk 5]
```diff
@@ -1,225 +1,352 @@
...
 
-Examples: `T add n/Assignment 1 due d/20/11/2020 a/NUS w/20 r/Focus on Chapters 1-3`
+Examples: `T add n/Assignment 1 d/20/11/2020 a/NUS w/20 r/Focus on Chapters 1-3`
+
+Remark: The `DATE` provided must be in the form `dd/MM/yyyy`. The `WEIGHTAGE` provided must be a number.
```
Comment (UncleGrandpa925) [inline docs/UserGuide.md]: Should we merge location into remark now?

File: docs/UserGuide.md [hunk 6]
```diff
@@ -1,225 +1,333 @@
...
 
-</div>
+## Features <a name="features"></a>
 
-## Module
+## Module <a name="module"></a>
 
 ### View
-For: Shows the information about a module.
+For: Show information about the module with the specified `MODULE_CODE`.
```
Comment (UncleGrandpa925) [inline docs/UserGuide.md]: Should it be to switch to the module tab now?

File: docs/UserGuide.md [hunk 7]
```diff
@@ -1,225 +1,333 @@
...
+
+Remark: The `MODULE_CODE` provided must be present in the Module list.
 
 ### Add
 
 For: Add a new module.
 
-Format: `M add m/{module_code} n/{name} d/{description}`
-Examples: `M add m/CS1231S n/Discrete Structures d/Introductory mathematical tools required for computer science`
+Format: `M add m/MODULE_CODE n/NAME`
```
Comment (UncleGrandpa925) [inline docs/UserGuide.md:122]: love this simplicity 👍

File: docs/UserGuide.md [hunk 8]
```diff
@@ -1,225 +1,333 @@
...
-Format: `T add n/{task name} d/{task date} a/{location} w/{weightage} r/{remark}`
+Examples: `L add m/CS2103T n/tutorial d/2pm - 3pm Wed a/COM1`
 
-Examples: `T add n/Assignment 1 due d/20/11/2020 a/NUS w/20 r/Focus on Chapters 1-3`
+Remark: The module must exist prior to adding a lesson. `DATE` must be of the form `START_TIME - END_TIME DAY_OF_WEEK`.`TYPE` must be one of
+ the following:
+* `lecture`
+* `tutorial`
+* `lab`
+* `sectional`
```
Comment (UncleGrandpa925) [inline docs/UserGuide.md:152]: there is recitation as well!

File: docs/UserGuide.md [hunk 9]
```diff
@@ -1,225 +1,333 @@
...
 
-Examples: `T list`
+Remark: The `INDEX` provided must be positive and cannot be larger than the length of the Lesson list.
 
-## Lesson
+## Task <a name="task"></a>
 
 ### View
-For: Shows the information about a lesson
+For: Show the information about a task.
```
Comment (UncleGrandpa925) [inline docs/UserGuide.md]: actually it will like expand the task (view in greater details)

File: docs/UserGuide.md [hunk 10]
```diff
@@ -1,225 +1,333 @@
...
 
-Format: `L info m/{module_code} t/{type}`
+Format: `T info INDEX`
```
Comment (UncleGrandpa925) [inline docs/UserGuide.md]: Now I think we should rename it to view

File: docs/UserGuide.md [hunk 11]
```diff
@@ -1,225 +1,333 @@
...
+For: Add a new task.
 
-Format: `L add m/{module_code} n/{type} t/{time} v/{venue or Zoom link}`
+Format: `T add n/NAME d/DATE [m/MODULE_CODE] [r/REMARK]`
 
-`{time}` : the time of the first lesson of that type in this semester
+Examples: `T add n/Assignment 1 d/20/11/2020 r/Focus on Chapters 1-3`
 
-Examples: `L add m/CS2103T n/tutorial t/01/08/2019 4pm-6pm v/COM1`
+Remark: A task can either belong to a module or not. Hence, the `MODULE_CODE` is optional. The `DATE` provided must
```
Comment (UncleGrandpa925) [inline docs/UserGuide.md:192]: DATE with hours:minutes if they like as well?

**Review by UncleGrandpa925 — CHANGES_REQUESTED** [review]
Thanks for writing this detailed UG! I have left some minor comments!

File: docs/UserGuide.md [hunk 12]
```diff
@@ -1,225 +1,342 @@
...
+Examples: `L add m/CS2103T n/tutorial d/2pm - 3pm Wed a/COM1`
 
-Examples: `L add m/CS2103T n/tutorial t/01/08/2019 4pm-6pm v/COM1`
+Remark: The module must exist prior to adding a lesson. `DATE` must be of the form `START_TIME - END_TIME DAY_OF_WEEK`.`TYPE` must be one of
+ the following:
+* `lecture`
+* `tutorial`
+* `lab`
+* `sectional`
+* `recitation`
```
Comment (UncleGrandpa925) [inline docs/UserGuide.md:160]: for this, is it better if we have the type like lec (for lecture), tut(for tutorial)?

File: docs/UserGuide.md [hunk 13]
```diff
@@ -1,225 +1,342 @@
...
 
 For: Edit the detail of a lesson
 
-Format: `L edit m/{module_code} [n/{type}] [t/{time}] [v/{venue or Zoom link}]`
+Format: `L edit INDEX [m/MODULE_CODE] [n/TYPE] [d/DATE] [a/LOCATION]`
+
+Examples: `L edit 1 m/CS2103T n/tutorial d/2pm - 3pm Monday`
 
-Examples: `L edit m/CS2103T n/tutorial t/02/08/2019 4pm-6pm v/COM1`
+Remark: The `INDEX` provided must be positive and cannot be larger than the length of the Lesson list. At least 1
```
Comment (UncleGrandpa925) [inline docs/UserGuide.md]: "the length of the Lesson list" sounds a little unnatural, maybe "the number of lessons"?

File: docs/UserGuide.md [hunk 14]
```diff
@@ -1,225 +1,342 @@
...
+ field must be provided for editing. i.e. `L edit 1` is not allowed.
 
 ### Delete
 For: Delete all lessons of that type
```
Comment (UncleGrandpa925) [inline docs/UserGuide.md]: not of that type right? now we changed to deleting by index already

File: docs/UserGuide.md [hunk 15]
```diff
@@ -1,225 +1,342 @@
...
 
-Format: `L delete m/{module_code} n/{type}`
+Format: `L delete INDEX`
 
-Examples: `L delete m/CS2103T n/tutorial`
+Examples: `L delete 1`
 
+Remark: The `INDEX` provided must be positive and cannot be larger than the length of the Lesson list.
```
Comment (UncleGrandpa925) [inline docs/UserGuide.md]: same here, maybe "the number of lessons"?

**Review by UncleGrandpa925 — CHANGES_REQUESTED** [review]

File: docs/UserGuide.md (hunk 12 above)
Comment (simonteozw) [reply docs/UserGuide.md:160]: I think we can just leave it for now

**Review by UncleGrandpa925 — APPROVED** [review]

**Pull request merged**